### PR TITLE
feat: support cjs and esm for cloudflare and edge common

### PR DESCRIPTION
**Pull request by yusinto**
* add example test app for cloudflare
* adds support for cjs and esm bundles for sdk-server-common and cloudflare

### CODE REVIEW

File: packages/sdk/cloudflare/example/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Example test app for Cloudflare LaunchDarkly SDK
+
+This is an example test app to showcase the usage of the Cloudflare LaunchDarkly
+SDK. This app was created with wrangler v2.
```
Comment (kinyoklion) [inline packages/sdk/cloudflare/example/README.md:4]: I know it says it was created with wrangler v2, but should we have a prerequisites/dependencies section?

File: packages/sdk/cloudflare/example/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects */
+    // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+    // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
+
+    /* Language and Environment */
+    "target": "es2021" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+    "lib": [
+      "es2021"
+    ] /* Specify a set of bundled library declaration files that describe the target runtime environment. */,
+    "jsx": "react" /* Specify what JSX code is generated. */,
```
Comment (kinyoklion) [inline packages/sdk/cloudflare/example/tsconfig.json:18]: Do we need jsx for some reason?

File: packages/sdk/cloudflare/example/wrangler.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name = "example"
+main = "dist/index.mjs"
+compatibility_date = "2023-04-02"
+compatibility_flags = [ "nodejs_compat" ]
+kv_namespaces = [{ binding = "LD_KV", id = "616aadb3c9474693bb9faaa4e53ea40c", preview_id = "88c3bd2abe5a4c3eb9b95829b86c2449" }]
```
Comment (kinyoklion) [inline packages/sdk/cloudflare/example/wrangler.toml:5]: Do these IDs have any meaning?

**Review by kinyoklion — APPROVED** [review]
Approved with comments.

File: packages/sdk/cloudflare/example/tsconfig.json (hunk 2 above)
Comment (yusinto) [reply packages/sdk/cloudflare/example/tsconfig.json:18]: This comes default from wrangler init. We don't need jsx in our example app, but maybe others may in their apps.

File: packages/sdk/cloudflare/example/wrangler.toml (hunk 3 above)
Comment (yusinto) [reply packages/sdk/cloudflare/example/wrangler.toml:5]: They refer to the kv ids in my cloudflare account. I should remove them and document them in the readme. Thanks for picking this up.

File: packages/sdk/cloudflare/example/README.md (hunk 1 above)
Comment (yusinto) [reply packages/sdk/cloudflare/example/README.md:4]: Done.

File: packages/sdk/cloudflare/example/wrangler.toml (hunk 3 above)
Comment (yusinto) [reply packages/sdk/cloudflare/example/wrangler.toml]: Done.

**Pull request merged**